disnake/state.py

```diff
@@ -930,6 +931,25 @@ def parse_message_reaction_remove_emoji(
                 if reaction:
                     self.dispatch("reaction_clear_emoji", reaction)
 
+    def _handle_poll_event(
+        self, raw: RawMessagePollVoteActionEvent, event_type: Literal["add", "remove"]
+    ) -> None:
+        message = self._get_message(raw.message_id)
+        guild = self._get_guild(raw.guild_id)
+        if guild is not None:
+            channel = guild.get_channel(raw.channel_id)
+            self.dispatch(f"message_poll_vote_{event_type}", guild, channel, message, raw.answer_id)
```

Comment:
Yeahh, I agree that this event has too many fields.
What about only having parameters for the user/member and the `PollAnswer`, and adding a `PollAnswer.poll: Poll` to be able to retrieve the poll object (and from there, you already automatically get message/channel/guild/etc.) from there?
Would be similar to how reaction events work, given `dispatch("reaction_add", reaction, user)` and `Reaction.message: Message`.

Reply:
We need a poll cache then. Passing the member to the user is not a problem, but passing a PollAnswer object requires caching polls and poll answers (which should already be cached if messages are cached)

Reply:
Polls should already be cached by the message cache, yea... I don't think a separate cache is useful here. Haven't tested it, but this should work to fetch the corresponding answer object here:
```py
if (message := self._get_message(raw.message_id)) and message.poll:
    answer = utils.get(message.poll.answers, id=raw.answer_id)
```

Reply:
Forgot for a moment that `utils.get` exist 😢
Should we pass the answer object to the raw event too? I'm already passing the member object (which is optional ofc) similarly to how we are doing with other events

Reply:
just the member object is fine, yeah. Don't really want to make the raw event too complex, people can always get stuff from cache manually if desired